src/ccd2rgb.jl

```diff
@@ -46,3 +51,64 @@ ccd2rgb(red::ImageHDU, green::ImageHDU, blue::ImageHDU; stretch = identity, shap
     ccd2rgb((read(red), WCS.from_header(read_header(red, String))[1]), (read(green), WCS.from_header(read_header(green, String))[1]),
             (read(blue), WCS.from_header(read_header(blue, String))[1]), stretch = stretch, shape_out = shape_out)
 
+
+
+function composechannels(
+    images,
+    multipliers=ones(size(images)), # 0.299 * R + 0.587 * G + 0.114 * B
+    channels=["#F00", "#0F0", "#00F"];
+    clims=extrema,
+    stretch=identity,
+    # reproject = all(==(wcs(first(images))), wcs(img) for img in images) ? false : wcs(first(images)),
+    reproject = false,
+    shape_out = size(first(images)),
+)
+    if reproject == false
+        reprojected = images
+    else
+        if reproject == true
+            reproject = first(images)
+        end
+        reprojected = map(images) do image
+            Reproject.reproject(image, reproject; shape_out)[1]
+        end
+    end
+    I = broadcast(+, reprojected...) ./ length(reprojected)
+    I .= (x -> stretch(x)/x).(I)
+
+    colors = parse.(Colorant, channels)
+    # @show colors
+        
+    # red_rp .*= I
+    # green_rp .*= I
+    # blue_rp .*= I
+    
+    # m1 = maximum(x->isnan(x) ? -Inf : x, red_rp)
+    # m2 = maximum(x->isnan(x) ? -Inf : x, green_rp)
+    # m3 = maximum(x->isnan(x) ? -Inf : x, blue_rp)
+    # return colorview(RGB, red_rp./m1 , green_rp./m2, blue_rp./m3)
+
+    # return colorview(RGB, (reprojected .* multipliers)...)
+
+    ## TODO: this all needs to be lazy
+
+    colorized = map(eachindex(reprojected)) do i
```

Comment:
Do you have to do this eagerly? MappedArrays would allow you to do it lazily, which might be nice for large images. (Though there are circumstances where lazy can cause trouble.)
```julia
mred = mappedarray(x -> r*x, redchannel)
...
colorview(RGB, mred, mgreen, mblue)
```
would be 100% lazy.